source/developer/oauth-2-0-applications.rst

```diff
@@ -71,12 +71,23 @@ Note: **Client Secret** can be regenerated by the application creator or System
 Granting permissions to your application 
 ---------------------------------------------------------
 
-Once you have created an OAuth 2.0 application, all users on the Mattermost server are automatically given access to it. Deleting the application will revoke access from all users, and only the user who created the application or a System Administrator can delete the app.
+Once you have created an OAuth 2.0 application, all users on the Mattermost server are automatically given access to it. 
 
-If the application was set up as a trusted OAuth 2.0 app on the Mattermost server, authorization from users is not required. Otherwise, the following page will be provided to accept or deny authorization when authenticating the app for the first time.
+The application does not automatically have permissions based on the user who registers the app - the permissions are based on which users go through the OAuth flow, which could be the user who registers the app or anyone else on the system.
+
+If the application was registered as a trusted OAuth 2.0 app on the Mattermost server, authorization from users is not required. Otherwise, the following page will be provided to accept or deny authorization when authenticating the app for the first time.
 
 .. image:: ../images/oauth2_authorization_screen.PNG
 
+Once authorized, the application receives an access token to perform requests on behalf of that user and has full access to perform any action that the user has permissions to.
```

Comment:
nit: trailing with a preposition can be awkward. Perhaps, `... on behalf of that user. The application can then perform any action for which the user has permission."

Reply:
Good catch! Updated